pkg/resources/common/validation.go

```diff
@@ -58,3 +61,35 @@ func ValidateRules(rules []rbacv1.PolicyRule, isNamespaced bool, fldPath *field.
 	}
 	return returnErr
 }
+
+func CheckCreatorPrincipalName(userCache controllerv3.UserCache, obj metav1.Object) (*field.Error, error) {
+	annotations := obj.GetAnnotations()
+	principalName := annotations[auth.CreatorPrincipalNameAnn]
+	if principalName == "" { // Nothing to check.
+		return nil, nil
+	}
+
+	annotationsFieldPath := field.NewPath("metadata").Child("annotations")
+
+	creatorID := obj.GetAnnotations()[auth.CreatorIDAnn]
+	if creatorID == "" {
+		return field.Invalid(annotationsFieldPath, auth.CreatorPrincipalNameAnn, fmt.Sprintf("annotation %s is required", auth.CreatorIDAnn)), nil
+	}
+
+	user, err := userCache.Get(creatorID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return field.Invalid(annotationsFieldPath, auth.CreatorPrincipalNameAnn, fmt.Sprintf("creator user %s doesn't exist", creatorID)), nil
+		}
+		return nil, fmt.Errorf("error getting creator user %s: %w", creatorID, err)
+	}
+
+	creatorPrincipalName := annotations[auth.CreatorPrincipalNameAnn]
+	for _, principal := range user.PrincipalIDs {
+		if principal == creatorPrincipalName {
+			return nil, nil
+		}
+	}
+
+	return field.Invalid(annotationsFieldPath, auth.CreatorPrincipalNameAnn, fmt.Sprintf("creator user %s doesn't have principal %s", creatorID, creatorPrincipalName)), nil
```

Comment:
We already have the creatorPrincipalName from L67:

```go
principalName := annotations[auth.CreatorPrincipalNameAnn]
```

```suggestion
	for _, principal := range user.PrincipalIDs {
		if principal == principalName {
			return nil, nil
		}
	}

	return field.Invalid(annotationsFieldPath, auth.CreatorPrincipalNameAnn, fmt.Sprintf("creator user %s doesn't have principal %s", creatorID, principalName)), nil
```

Reply:
Duh, of course. The no-op check came later.